hw8/hw8_t01.py

```diff
@@ -0,0 +1,38 @@
+# noqa: D208
+"""
+We have a file that works as key-value storage, each like is represented as key.
+
+ and value separated by = symbol, example:
+
+name=kek last_name=top song_name=shadilay power=9001
+
+Values can be strings or integer numbers.
+If a value can be treated both as a number and a string, it is treated as number.
+
+Write a wrapper class for this key value storage that works like this:
+
+storage = KeyValueStorage('path_to_file.txt') that has its keys and values accessible
+as collection items and as attributes.
+ Example: storage['name'] # will be string 'kek' storage.song_name # will be
+ 'shadilay' storage.power # will be integer 9001
+
+In case of attribute clash existing built-in attributes take precedence.
+ In case when value cannot be assigned to an attribute
+(for example when there's a line 1=something) ValueError should be raised.
+ File size is expected to be small, you are permitted to read it entirely into memory.
+ """
+
+
+class KeyValueStorage:  # noqa: D101
+    def __init__(self, path: str):
+        with open(path, "r") as file:
+            existing_builtins = KeyValueStorage.__dict__.keys()
```

Comment:
What is the type of this variable?

Reply:
The type is <class 'dict_keys'>

Reply:
What does this type mean?

Reply:
It is a class consisting of the dict keys. Judging by its output - it is a list type. (not good for `in` search?)

Reply:
> Judging by its output - it is a list type. 

Some time ago, a couple of decades, I have used a dial-up connection to the internet. This was a night tariff, so I have no internet access during the daytime.   Since that, I change my connection to 24/7.  And what is your excuse for not looking into the documentation in the python.org?

Reply:
https://www.python.org/dev/peps/pep-3106/ This page is hard to find on the web. Well - it is a set. Why shouldn't it be indeed?

Reply:
It is obvious to start from here: https://docs.python.org/3/library/stdtypes.html?highlight=keys#dict.keys

And later on, you will get here:  https://docs.python.org/3/library/stdtypes.html?highlight=keys#dict-views

The next question why I want you to read this description?

Reply:
`The objects returned by dict.keys(), dict.values() and dict.items() are view objects. They provide a dynamic view on the dictionary’s entries, which means that when the dictionary changes, the view reflects these changes.`
If Class __dict__ will change, so will do my `existing_builtins`. I thought that should not be an issue on init stage. I can use frozenset instead.

Reply:
The main idea is that this is just an alias so you could replace `if key not in existing_builtins:` to `if key not in KeyValueStorage.__dict__:`